src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -894,6 +920,14 @@ private boolean handleJmxNotificationForCassandra21(
                     .withId(segmentId)
                     .build());
 
+            // Since we can get out of order notifications,
+            // we need to exit if we already got the COMPLETE notification.
```

Comment:
can we add an `assert` here that failed hasn't already been marked true.

Reply:
`failedNotified` or `segmentFailed` ?

In any case, what's the point ? There would be a big fat bug in Cassandra if we could get a `SESSION_FAILED` notif and then a `SESSION_SUCCESS`. 
It shouldn't be up to Reaper to assert here, as it's not anymore a matter of race conditions (which Reaper can/should handle). 

Reply:
that's the point of assertions, to assert something we presume is true.
it also serves as documentation (another contributor to reaper may not know that about Cassandra)